bundles/org.eclipse.epsilon.rdf.emf/src/org/eclipse/epsilon/rdf/emf/RDFGraphResourceUpdate.java

```diff
@@ -306,27 +293,120 @@ private void addToBag(Object values, Bag container) {
 			container.add(values);
 		}
 
-		if (CONSOLE_OUTPUT_ACTIVE) {
-			reportContainer("After add to container ", container);
-		}
+		if (CONSOLE_OUTPUT_ACTIVE) {reportContainer("After add to container ", container);}
 	}
 	
-	private RDFList newList(Model model, EObject onEObject, EAttribute eAttribute, Object values) {		
-		// SUBJECT
-		Resource subjectNode = rdfGraphResource.getRDFResource(onEObject);
-		// PREDICATE
-		Property property = getProperty(eAttribute);
-		// OBJECT
-		RDFList objectNode = createRDFListOnModel(values, model);
-		model.add(subjectNode, property, objectNode);
+	private Seq newSequence(Model model, EObject onEObject, EStructuralFeature eStructuralFeature) {
+		Seq objectNode = model.createSeq();
+		model.add(createStatement(model, onEObject, eStructuralFeature, objectNode));
+		return objectNode;
+	}
+	
+	private Bag newBag(Model model, EObject onEObject, EStructuralFeature eStructuralFeature) {
+		Bag objectNode = model.createBag();
+		model.add(createStatement(model, onEObject, eStructuralFeature, objectNode));
 		return objectNode;
 	}
 	
-	private void addToList(Object values, RDFList container, int position, EAttribute eAttribute, EObject onEObject) {	
+	//
+	// List statement operations	
+	
+	private void checkAndRemoveEmptyList(RDFList container, EObject onEObject, EStructuralFeature eStructuralFeature) {
+		Model model = container.getModel();
+
+		if (!container.isValid()) {
+			if (CONSOLE_OUTPUT_ACTIVE) {System.out.println("Removing invalid (empty) container:" + container.asResource());}
+			Statement stmtToRemove = createStatement(model, onEObject, eStructuralFeature, container.asResource());
+			model.remove(stmtToRemove);
+			return;
+		}
+		
+		if (1 == container.size() && SINGLE_MULTIVALUES_AS_STATEMENT) {
+			// TODO convert list with 1 item to single statement
+			if (CONSOLE_OUTPUT_ACTIVE) {System.out.println("Removing container with 1 item and making a statement:" + container.asResource());}
+			RDFNode firstValue = container.iterator().next();
+			container.removeList();
+			model.remove(createStatement(model, onEObject, eStructuralFeature, container.asResource()));
+			model.add(createStatement(model, onEObject, eStructuralFeature, firstValue));
+			return;
+		}
+		
+		if (container.isEmpty()) {
+			if (CONSOLE_OUTPUT_ACTIVE) {System.out.println("Removing empty container:" + container.asResource());}
+			container.removeList();
+			model.remove (createStatement(model, onEObject, eStructuralFeature, container.asResource()));
+			return;
+		}
+	}
+	
+	private void headSafeRemoveFromList(RDFList container, RDFNode rdfNode) {
+		// TODO consider creating an RDFList wrapper class to abstract away these complexities (e.g. JenaList)
+		// Fixes the blank node for the list when value is removed from the head
+		RDFList newContainer;
+		newContainer = container.remove(rdfNode);
+		if (!newContainer.equals(container)) {
```

Comment:
Note: I'll try to see if we can avoid having to do this. It feels more natural to just change the statement from the EObject's resource that points to the list.

Reply:
I've pushed a change that passes all tests without needing this workaround. @OwenR-York can you review it?